okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -527,19 +544,16 @@ public void close() {}
           executor.execute(clientFrameHandler);
         }
 
-        FrameWriter rawFrameWriter;
         synchronized (lock) {
           socket = Preconditions.checkNotNull(sock, "socket");
+          frameWriter.becomeConnected(socket);
           maxConcurrentStreams = Integer.MAX_VALUE;
           startPendingStreams();
           if (sslSession != null) {
             securityInfo = new InternalChannelz.Security(new InternalChannelz.Tls(sslSession));
           }
         }
 
-        rawFrameWriter = variant.newWriter(sink, true);
-        frameWriter.becomeConnected(rawFrameWriter, socket);
-
         try {
           // Do these with the raw FrameWriter, so that they will be done in this thread,
```

Comment:
This doesn't look like it will work any more. It looks like they may need to be done directly within start().